genesyscloud/telephony_providers_edges_site_outbound_route/resource_genesyscloud_telephony_providers_edges_site_outbound_route.go

```diff
@@ -0,0 +1,247 @@
+				}
+
+				outboundRoutesSet.Add(dOutboundRoute)
+			}
+
+			_ = d.Set("outbound_routes", outboundRoutesSet)
+		} else {
+			_ = d.Set("outbound_routes", nil)
+		}
+
+		return cc.CheckState(d)
+	})
+}
+
+func updateSiteOutboundRoutes(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if exists := featureToggles.OutboundRoutesToggleExists(); !exists {
+		return util.BuildDiagnosticError(resourceName, fmt.Sprintf("Environment variable %s not set", featureToggles.OutboundRoutesToggleName()), fmt.Errorf("environment variable %s not set", featureToggles.OutboundRoutesToggleName()))
+	}
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getSiteOutboundRouteProxy(sdkConfig)
+
+	outboundRoutes := buildOutboundRoutes(d.Get("outbound_routes").(*schema.Set))
+
+	// Get the current outbound routes
+	outboundRoutesAPI, resp, err := proxy.getSiteOutboundRoutes(ctx, d.Id())
+	if err != nil {
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to get outbound routes for site %s error: %s", d.Id(), err), resp)
+	}
+
+	// Delete unwanted outbound routes first to free up classifications assigned to them
+	for _, outboundRouteFromAPI := range *outboundRoutesAPI {
+		// Delete route if no reference to it
+		if _, ok := nameInOutboundRoutes(*outboundRouteFromAPI.Name, *outboundRoutes); !ok {
+			resp, err := proxy.deleteSiteOutboundRoute(ctx, d.Id(), *outboundRouteFromAPI.Id)
+			if err != nil {
+				if util.IsStatus404(resp) {
+					return nil
+				}
+				return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to delete outbound route from site %s error: %s", d.Id(), err), resp)
+			}
+		}
+	}
+	time.Sleep(2 * time.Second)
+
+	// Update the outbound routes
+	for _, outboundRoute := range *outboundRoutes {
+		if outboundRouteUpdate, ok := nameInOutboundRoutes(*outboundRoute.Name, *outboundRoutesAPI); ok {
+			// Update the outbound route
+			outboundRouteUpdate.Name = outboundRoute.Name
+			outboundRouteUpdate.Description = outboundRoute.Description
+			outboundRouteUpdate.ClassificationTypes = outboundRoute.ClassificationTypes
+			outboundRouteUpdate.Enabled = outboundRoute.Enabled
+			outboundRouteUpdate.Distribution = outboundRoute.Distribution
+			outboundRouteUpdate.ExternalTrunkBases = outboundRoute.ExternalTrunkBases
+
+			_, resp, err := proxy.updateSiteOutboundRoute(ctx, d.Id(), *outboundRouteUpdate.Id, outboundRouteUpdate)
+			if err != nil {
+				return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to update outbound route with id %s for site %s error: %s", *outboundRouteUpdate.Id, d.Id(), err), resp)
+			}
+		} else {
+			// Add the outbound route if not already present
+			_, resp, err := proxy.createSiteOutboundRoute(ctx, d.Id(), &outboundRoute)
+			if err != nil {
+				return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to add outbound route to site %s error: %s", d.Id(), err), resp)
+			}
+		}
+	}
+
```

Comment:
This function is very big. I'd recommend having a function that returns the three lists - routes to create, update, and delete. And then calling functions to perform those operations. It could all go behind the proxy as well